lib/os/cbprintf_nano.c

```diff
@@ -37,77 +37,72 @@ typedef uint32_t uint_value_type;
 BUILD_ASSERT(sizeof(uint_value_type) <= 8U,
 	     "DIGITS_BUFLEN formula may be incorrect");
 
+#define OUTC(_c) do { \
+	out((int)(_c), ctx); \
+	if (IS_ENABLED(CONFIG_CBPRINTF_LIBC_SUBSTS)) { \
+		++count; \
+	} \
+} while (false)
+
 static void print_digits(cbprintf_cb out, void *ctx, uint_value_type num,
 			 unsigned int base, bool pad_before, char pad_char,
 			 int min_width, size_t *countp)
 {
 	size_t count = 0;
 	char buf[DIGITS_BUFLEN];
-	unsigned int i;
```

Comment:
Why is this being switched to a signed int?  The value should never be negative.